lms/static/js/student_account/views/account_settings_factory.js

```diff
@@ -0,0 +1,154 @@
+                                    model: accountSettingsModel,
+                                    title: gettext("Year of Birth"),
+                                    valueAttribute: "year_of_birth",
+                                    options: fields_data['year_of_birth']['options'],
+                                })
+                            },
+                            {
+                                view: new AccountSettingsFieldViews.DropdownFieldView({
+                                    model: accountSettingsModel,
+                                    title: gettext("Country"),
+                                    valueAttribute: "country",
+                                    options: fields_data['country']['options'],
+                                })
+                            },
+                            {
+                                view: new AccountSettingsFieldViews.DropdownFieldView({
+                                    model: accountSettingsModel,
+                                    title: gettext("Preferred Language"),
+                                    valueAttribute: "language",
+                                    helpMessage: gettext("If your preferred language isn't available as an edX language, you can indicate your preferance here."),
+                                    options: fields_data['preferred_language']['options'],
+                                })
+                            },
+                        ]
+                    },
+                    {
+                        title: gettext("Connected Accounts"),
+                        fields: [
+                            {
+                                view: new AccountSettingsFieldViews.LinkFieldView({
+                                    model: accountSettingsModel,
+                                    title: gettext("Facebook"),
+                                    valueAttribute: 'auth-facebook',
+                                    linkTitle: gettext("Link"),
+                                    helpMessage: gettext("Under construction."),
+                                })
+                            },
+                            {
+                                view: new AccountSettingsFieldViews.LinkFieldView({
+                                    model: accountSettingsModel,
+                                    title: gettext("Google"),
+                                    valueAttribute: 'auth-google',
+                                    linkTitle: gettext("Link"),
+                                    helpMessage: gettext("Under construction."),
+                                })
+                            },
+                        ]
+                    }
+                ]
+            });
+
+            accountSettingsView.render();
+            accountSettingsModel.fetch().done(function (data, textStatus, jqXHR) {
```

Comment:
What happens if the fetch fails? I think there needs to be some logic to show an error message.